### DISCUSSION TITLE
Trouble installing netbox

### DISCUSSION THREAD

**Discussion by saltymedic** [netbox-community/netbox#6404] [Q&A] [answered]
I followed the installation instructions from
https://netbox.readthedocs.io/en/stable/installation/

The only difference I noted is that there was an extra /netbox directory level that showed up even though I followed the directions. This is on Ubuntu 20.04

I got to the end of the Gunicorn instructions
https://netbox.readthedocs.io/en/stable/installation/4-gunicorn/

When I run systemctl status netbox, this is what it shows me:

`● netbox.service - NetBox WSGI Service
     Loaded: loaded (/etc/systemd/system/netbox.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Wed 2021-05-12 11:09:14 MDT; 2s ago
       Docs: https://netbox.readthedocs.io/en/stable/
    Process: 35575 ExecStart=/opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbo>
   Main PID: 35575 (code=exited, status=203/EXEC)
`
When I run journalctl -eu netbox, this is what I get:

`May 12 11:12:16 rlc-office systemd[1]: netbox.service: Main process exited, code=exited, status=203/EXEC
May 12 11:12:16 rlc-office systemd[1]: netbox.service: Failed with result 'exit-code'.
May 12 11:12:46 rlc-office systemd[1]: netbox.service: Scheduled restart job, restart counter is at 1456.
May 12 11:12:46 rlc-office systemd[1]: Stopped NetBox WSGI Service.
May 12 11:12:46 rlc-office systemd[1]: Started NetBox WSGI Service.
May 12 11:12:46 rlc-office systemd[35731]: netbox.service: Failed to execute command: No such file or directory
May 12 11:12:46 rlc-office systemd[35731]: netbox.service: Failed at step EXEC spawning /opt/netbox/venv/bin/gunico>
May 12 11:12:46 rlc-office systemd[1]: netbox.service: Main process exited, code=exited, status=203/EXEC
May 12 11:12:46 rlc-office systemd[1]: netbox.service: Failed with result 'exit-code'.
May 12 11:13:16 rlc-office systemd[1]: netbox.service: Scheduled restart job, restart counter is at 1457.
May 12 11:13:16 rlc-office systemd[1]: Stopped NetBox WSGI Service.
May 12 11:13:16 rlc-office systemd[1]: Started NetBox WSGI Service.
May 12 11:13:16 rlc-office systemd[35755]: netbox.service: Failed to execute command: No such file or directory
May 12 11:13:16 rlc-office systemd[35755]: netbox.service: Failed at step EXEC spawning /opt/netbox/venv/bin/gunico>
May 12 11:13:16 rlc-office systemd[1]: netbox.service: Main process exited, code=exited, status=203/EXEC
May 12 11:13:16 rlc-office systemd[1]: netbox.service: Failed with result 'exit-code'.
`

Any idea on how to get this going?

Thanks for your help

Jason

**Comment by candlerb** [comment, marked as answer]
> The only difference I noted is that there was an extra /netbox directory level that showed up even though I followed the directions.

Please explain exactly what you mean by that.

You should unpack or git clone in /opt/netbox.  There will indeed be other subdirectories called "netbox"; the configuration goes in `/opt/netbox/netbox/netbox/configuration.py` (yes, that's an awkward directory name)

Your top level /opt/netbox directory should look something like this:

```
root@netbox:~# ls /opt/netbox
CHANGELOG.md     NOTICE                 contrib      local_requirements.txt  netbox            upgrade.sh
CONTRIBUTING.md  README.md              docs         log                     requirements.txt  venv
LICENSE.txt      base_requirements.txt  gunicorn.py  mkdocs.yml              scripts
```

> status=203/EXEC

[systemd exit code](https://freedesktop.org/software/systemd/man/systemd.exec.html#Process%20Exit%20Codes) 203 means: "The actual process execution failed (specifically, the execve(2) system call). Most likely this is caused by a missing or non-accessible executable file."

Perhaps `/opt/netbox/venv/bin/gunicorn` does not exist, or it isn't accessible?  Did you run the `./upgrade.sh` script which creates the virtualenv in the /opt/netbox/venv directory?

**Reply by saltymedic** [reply]
Per the directions here:

[https://netbox.readthedocs.io/en/stable/installation/3-netbox/](url)


I did:

`sudo mkdir -p /opt/netbox/ && cd /opt/netbox/`

and then 

`sudo git clone -b master https://github.com/netbox-community/netbox.git`

Following those directions, it put the configuration.py file in:

`/opt/netbox/netbox/netbox/netbox/`

I realized based on that  that there seemed to be an extra netbox directory from what I was seeing in the directions afterwards

Maybe I should have run the git command from the /opt directory?

my `/opt/netbox/netbox` directory looks like:

`jcotting@rlc-office:/opt/netbox/netbox$ ls`

`base_requirements.txt  contrib          docs         LICENSE.txt  netbox  README.md         scripts     venv
CHANGELOG.md           CONTRIBUTING.md  gunicorn.py  mkdocs.yml   NOTICE  requirements.txt  upgrade.sh`

gunicorn does exist, perhaps this extra directory thing puts it in a place its not looking for it?

`jcotting@rlc-office:/opt/netbox/netbox/venv/bin$ ll`
`total 104`
`drwxr-xr-x 3 root root 4096 May 11 08:44 ./`
`drwxr-xr-x 6 root root 4096 May 11 08:41 ../`
`-rw-r--r-- 1 root root 2203 May 11 08:41 activate`
`-rw-r--r-- 1 root root 1255 May 11 08:41 activate.csh`
`-rw-r--r-- 1 root root 2407 May 11 08:41 activate.fish`
`-rw-r--r-- 1 root root 8834 May 11 08:41 Activate.ps1`
`-rwxr-xr-x 1 root root  240 May 11 08:43 chardetect*`
`-rwxr-xr-x 1 root root  282 May 11 08:43 django-admin*`
`-rwxr-xr-x 1 root root  672 May 11 08:43 django-admin.py*`
`-rwxr-xr-x 1 root root  249 May 11 08:41 easy_install*`
`-rwxr-xr-x 1 root root  249 May 11 08:41 easy_install-3.8*`
`-rwxr-xr-x 1 root root  236 May 11 08:44 gunicorn*`

Looking back through my .bash_history file it does show that I ran ./upgrade.sh

`jcotting@rlc-office:/opt/netbox/netbox/venv$ ll`
`total 28`
`drwxr-xr-x 6 root root 4096 May 11 08:41 ./`
`drwxr-xr-x 9 root root 4096 May 11 22:36 ../`
`drwxr-xr-x 3 root root 4096 May 11 08:44 bin/`
`drwxr-xr-x 2 root root 4096 May 11 08:41 include/`
`drwxr-xr-x 3 root root 4096 May 11 08:41 lib/`
`lrwxrwxrwx 1 root root    3 May 11 08:41 lib64 -> lib/`
`-rw-r--r-- 1 root root   69 May 11 08:41 pyvenv.cfg`
`drwxr-xr-x 3 root root 4096 May 11 08:41 share/`

Should I try removing everything and run git from the /opt directory?

Thank you!

Jason

**Reply by candlerb** [reply]
You misread the instructions.  They [actually say](https://netbox.readthedocs.io/en/stable/installation/3-netbox/#option-b-clone-the-git-repository):

```
sudo git clone -b master https://github.com/netbox-community/netbox.git .
```

There is a dot by itself on the end of the line.  This clones into the current directory.

Admittedly it's not obvious when reading, but it would work if you copy-pasted it exactly.  It could perhaps have been written more clearly as

```
sudo git clone -b master https://github.com/netbox-community/netbox.git ./
```

**Reply by saltymedic** [reply]
That was it. That pesky little period.

Thank you very very much for your help. Its working now.

Jason
